04_Scheduling/03_Tasks_And_Threads.md

```diff
@@ -0,0 +1,132 @@
+# Tasks and Threads
+
+** THIS SECTION IS IN EARLY STAGES For now it will be more a set of bullet lists with things to be expaneded in the future ** 
+
+## Tasks
+
+### What are tasks
+
+In a task usually there are the following information: 
+
+* Virtual memory base
+* Task ID
+* Thread List (if we implement multithreading) or function pointer if we don't
+* Stack reference
+* Arguments 
+* Parent task
+
+While a thread should contains at minimum:
+
+* Thread name
+* Thread id
+* Stack pointer
+* Execution frame
+* Pointer to function to run
+* Eventually arguments for the function
+* Depending on how the scheduler is designed, it could contains a pointer to the next thread
+* Task containing the current thread
+
+### Prerequisites 
+
+Even if technically not necessary, is better to have implemented a memory allocation mechanism, ideally both physical and virtual memory manager should be implemented, if we want to have full tasks/thread separation and protection. But in case we don't want to have memory separation and protection in place a physical memory manager can be enough.
```

Comment:
While possible to implement scheduling without a vmm (only using pmm), I don't think it's an idea we should encourage. It'll likely lead to bad design decisions (tasks allocating physical memory instead of working with virtual addresses), and also implies that they dont have a heap.
I think it's safe to assume that the reader is reading things in order, and has therefore read the memory/allocators section.